### DISCUSSION TITLE
关于字体替换逻辑

### DISCUSSION THREAD

**Discussion by aprz512** [chinosk6/gakuen-imas-localify#34] [Q&A] [answered]
从代码上看，这个替换应该是对所有游戏都有效才对。
我想汉化另一个游戏《六狱恋歌》，但是替换字体功能没有生效。
有群可以讨论一下么？

**Comment by chinosk6** [comment]
本游戏只涉及 `TextMeshPro`，且保留了 `UnityEngine.Font::CreateFontFromPath` 这个方法，所以目前的代码即可。
除了 TextMesh，可能还有其它组件比如 `UnityEngine.UI.Text` 用于显示文本。
你需要先确认你的游戏具体使用了哪些组件来显示文本。
而且在打包过程中，不一定所有游戏都会保留 `UnityEngine.Font::CreateFontFromPath`。如果没有，本插件的方法也不可行，需要通过 AssetBundle 加载字体。所以不一定每个游戏的方法都通用。
群的话，目前只有一个插件大群: 975854705

**Reply by aprz512** [reply]
好的，感谢。
我在TMP_Text_PopulateTextBackingArray里面打印了日志，对话文本显示时确实是走了这里。
也dump了 il2cpp 里面的方法结构，用 ida 打开也能看到 Internal_CreateFontFromPath 这个方法是存在的。
所以感觉比较奇怪。

**Comment by aprz512** [comment]
加载 assetbundle 的逻辑跑通了，但是出现了所有文字全变成   口    的问题，请问这个有遇到过吗？

**Reply by chinosk6** [reply, marked as answer]
通过 AssetBundle 加载字体的话，[Trainers-Legend-G](https://github.com/MinamiChiwa/Trainers-Legend-G) 和 [scsp-localify](https://github.com/chinosk6/scsp-localify) 这两个 repo 都可以参考

**Reply by aprz512** [reply]
感谢
